src/plugins/image/display/image-display.js

```diff
@@ -9,18 +9,37 @@ import GithubFlavoredMarkdown from '../../../common/github-flavored-markdown.js'
 import { ReactCompareSlider, ReactCompareSliderImage } from 'react-compare-slider';
 
 function ImageDisplay({ content }) {
+  const clipEffectImageRef = useRef();
+  const clipEffectCanvasRef = useRef();
   const maxWidth = content.maxWidth || 100;
   const { text, sourceType, sourceUrl, effect } = content;
 
   const clientConfig = useService(ClientConfig);
   const githubFlavoredMarkdown = useService(GithubFlavoredMarkdown);
+  const src = getImageSource(clientConfig.cdnRootUrl, sourceType, sourceUrl);
+
+  useEffect(() => {
+    if (effect?.type !== EFFECT_TYPE.clip) {
+      return;
+    }
+    const img = clipEffectImageRef.current;
+    const canvas = clipEffectCanvasRef.current;
+    const context = canvas.getContext('2d');
+    const width = img.naturalWidth * (effect.region.width / 100);
+    const height = img.naturalHeight * (effect.region.height / 100);
+    const x = img.naturalWidth * (effect.region.x / 100);
+    const y = img.naturalHeight * (effect.region.y / 100);
+    canvas.width = width;
+    canvas.height = height;
+    context.drawImage(img, x, y, width, height, 0, 0, width, height);
+  }, [clipEffectCanvasRef, effect]);
```

Comment:
I think we have to specify `clipEffectImageRef.current` and `clipEffectCanvasRef.current`?

Reply:
Thanks for noticing this, added `clipEffectImageRef`.